CleanseTests/CycleTests.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  CycleTests.swift
+//  Cleanse
+//
+//  Created by Mike Lewis on 7/20/16.
+//  Copyright © 2016 Square, Inc. All rights reserved.
+//
+
+import Foundation
+
+
+import XCTest
+
+@testable import Cleanse
+
+class CycleTests: XCTestCase {
+
+    class AA {
+        init(b: BB) {
+
+        }
+    }
+
+    class BB {
+        init(a: AA) {
+
+        }
+    }
+
+    struct BadComponent : RootComponent {
+        typealias  Root = AA
+
+        static func configure<B : Binder>(binder binder: B) {
+
+            binder.bind(AA.self).asSingleton().to(factory: AA.init)
+            binder.bind(BB.self).asSingleton().to(factory: BB.init)
+        }
+    }
+
+
+
+    class AA_OK {
+        init(b: BB_OK) {
+
+        }
+    }
+
+    class BB_OK {
+        init(a: WeakProvider<AA_OK>) {
+        }
+    }
+
+    struct GoodComponent : RootComponent {
+        typealias  Root = AA
+
+        static func configure<B : Binder>(binder binder: B) {
+
+            binder.bind().asSingleton().to(factory: AA_OK.init)
+            binder.bind().asSingleton().to(factory: BB_OK.init)
+        }
+    }
+
+    func testCycleDetection() {
+        do {
+            _ = try ComponentFactory.of(BadComponent)
+            XCTFail("Should not succeed")
+        } catch let e {
+
```

Comment:
Yeah, I need to copy what we're doing in ErrorTest... which is what's going to happen before merging